### PR TITLE
Sprintthree

### PR DESCRIPTION
**Pull request by girishchandrad**
Hi Juan,

Please review.

thanks,
Girish Chandra

### CODE REVIEW

File: app/controllers/searches.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { alias } from '@ember/object/computed';
+
+
+export default Controller.extend({
+ queryParams: ['searchkey'],
+ searchkey:null,
+ keywordSearches: null,
+ resultLength: alias('model.length'),
+ searchKeyChange: alias('searchkey'),
+ keywordSearches: oneWay('searchKeyChange'),
```
Comment (juandsierra82) [inline app/controllers/searches.js]: why are you doing oneway binding? if you are concerned about the query to the back end, you can always trigger the query with an action. should for the most part keep two way binding.

File: app/controllers/searches.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ isDisabled: empty('keywordSearches'),
+
+  actions: {
+    setSearchParams(){
+        let keywordSearches = this.get('keywordSearches');
```
Comment (juandsierra82) [inline app/controllers/searches.js]: no need to declare a variable here.

File: app/controllers/searches.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { empty } from '@ember/object/computed';
+import { alias } from '@ember/object/computed';
+
+
+export default Controller.extend({
+ queryParams: ['searchkey'],
+ searchkey:null,
+ keywordSearches: null,
+ resultLength: alias('model.length'),
+ searchKeyChange: alias('searchkey'),
```
Comment (juandsierra82) [inline app/controllers/searches.js]: why are you using aliases here?

File: app/router.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const Router = EmberRouter.extend({
 
 Router.map(function() {
   this.route('form', { path: 'form/:id' });
-  this.route('search');
+  this.route('searches', { path: 'searches' });
```
Comment (juandsierra82) [inline app/router.js]: you don't need to declare a path if the path is the same as the path name

File: app/routes/searches.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+export default Route.extend({
+
+queryParams:{
+  searchkey: {
+    refreshModel: true
+  }
+},
+  model(params){
+    if (params.searchkey === "" ||  params.searchkey === undefined) {
+        return ""
```
Comment (juandsierra82) [inline app/routes/searches.js]: semi colon

File: app/styles/site-views/form-view.scss [hunk 6]
```diff
@@ -9,3 +9,17 @@
   font-size: $text-display-size;
    margin-top: $footer-display-margin-top;
 }
+
```
Comment (juandsierra82) [inline app/styles/site-views/form-view.scss:12]: .form-display-truncate should have nested children. please make sure you follow the scss guidelines for nesting.

**Review by juandsierra82 — COMMENTED** [review]
make sure you have that back button bug fixed (where i search main page, go to search page, change my search, click back, search again, my search should change again too)

File: app/controllers/searches.js (hunk 1 above)
Comment (girishchandrad) [reply app/controllers/searches.js]: Back button is fixed. one way binding is removed

File: app/controllers/searches.js (hunk 2 above)
Comment (girishchandrad) [reply app/controllers/searches.js]: removed

File: app/controllers/searches.js (hunk 3 above)
Comment (girishchandrad) [reply app/controllers/searches.js]: all aliases are removed

File: app/router.js (hunk 4 above)
Comment (girishchandrad) [reply app/router.js]: corrected

File: app/routes/searches.js (hunk 5 above)
Comment (girishchandrad) [reply app/routes/searches.js]: corrected

File: app/styles/site-views/form-view.scss (hunk 6 above)
Comment (girishchandrad) [reply app/styles/site-views/form-view.scss:12]: corrected

File: app/models/tax-form-due-date.js [hunk 7]
```diff
@@ -14,5 +15,12 @@ export default DS.Model.extend({
   dd: DS.attr('date'),
   fiscalDdDesc: DS.attr('string'),
   extDd: DS.attr('date'),
-  fiscalExtDdDesc: DS.attr('string')
+  fiscalExtDdDesc: DS.attr('string'),
+
+  ddDate:computed("dd", function(){
+    return moment(this.get("dd")).format("MMMM Do YYYY");
```
Comment (juandsierra82) [inline app/models/tax-form-due-date.js:21]: single quotes please

**Pull request merged**